python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -822,6 +830,100 @@ async def _wait_for_mtm1m3_detailed_state(
                 )
             self.log.debug(f"M1M3 detailed state {m1m3_detailed_state.detailedState!r}")
 
+    async def _wait_hard_point_test_ok(self, hp: int) -> None:
+        """Wait until the hard point test for the specified hard point
+        finishes.
+
+        Parameters
+        ----------
+        hp : `int`
+            Index of the hard point (starting from 1).
+
+        Raises
+        ------
+        RuntimeError
+            If the hp test failed.
+        """
+
+        self.log.info("Checking if the hard point breakaway test has passed.")
+
+        while True:
+            hp_test_state = MTM1M3.HardpointTest(
```

Comment:
Does that run? I bet what you need is hp_test_state = await self.rem.mtm1m3.evt_hardpointTestStatus.next(...).testState[hp-1] - drop MTM1M3HardpointTest please